src/uct/ib/dc/dc_mlx5.c

```diff
@@ -766,24 +766,15 @@ ucs_status_t uct_dc_device_query_tl_resources(uct_ib_device_t *dev,
 
 static inline ucs_status_t uct_dc_mlx5_iface_flush_dcis(uct_dc_mlx5_iface_t *iface)
 {
-    int is_flush_done = 1;
-    uct_dc_mlx5_ep_t *ep;
     int i;
 
     for (i = 0; i < iface->tx.ndci; i++) {
-        /* TODO: Remove this check - no need to wait for grant, because we
-         * use gc_list for removed eps */
-        if (!uct_dc_mlx5_iface_is_dci_rand(iface)) {
-            ep = uct_dc_mlx5_ep_from_dci(iface, i);
-            if ((ep != NULL) && uct_dc_mlx5_ep_fc_wait_for_grant(ep)) {
-                return UCS_INPROGRESS;
-            }
-        }
         if (uct_dc_mlx5_iface_flush_dci(iface, i) != UCS_OK) {
-            is_flush_done = 0;
+            return UCS_INPROGRESS;
```

Comment:
you stops on first incomplete flush.. is it ok? may be better to try to flush all other DCI's too?

Reply:
It is NOOP, we do not do anything with DCIs. Just checking whether we have outstanding ops. If at least one DCI has outstandings - flush is not completed